### PR TITLE
Also build sdist in CI

### PR DESCRIPTION
**Pull request by EpicWink**
So package can be built on platforms which don't have wheels

### CODE REVIEW

**Comment by bwoodsend** [comment]
Is this actually from personal experience of wanting to build on some weird platform or just mirroring the status quo? In my experience, most people installing from sdist are really just people building on normal but new platforms but with too old pip versions.

Normally, that wouldn't bother me but this project has build time dependencies on cython (you know that project that takes issue with every compiler ever) and numpy which doesn't ship precompiled wheels for anything particularly outlandish and is impractical to build from source. So to install this package on any non-tested platform (par musl which you're already addressing), you need to pre-install numpy either from source or via a native package manager, install the sdist, realise it's 15 minutes later that pip hasn't even touched this package because build isolation decided to build numpy from source, cancel and reinstall using pip's `--no-build-isolation` flag, realise you're missing cython, build cython from source, then reinstall this package and hope that cython doesn't have any nasty surprises for you.

So tl;dr, unless a *BSD user comes along and tells me otherwise, I'd prefer to just force said (hypothetical?) users to build from git.

**Comment by EpicWink** [comment]
I intended to use this package in Musl actually (see #29).

I'm my experience, it's rare that users want to use a NumPy-dependent library on a platform which NumPy doesn't support, and when they do, they know what they're getting into (and are fine with building NumPy from source, if they don't already have a cached wheel).

Also, it's good practice to provide the sdist, as redistributors prefer the sdist over the Git archive to build the package in their own environment.

Outside of Musl, I'm personally not too fussed, so you can close this PR if you want.

**Comment by bwoodsend** [comment]
In that case, I'll keep it as is.

I'm aware that Linux packages use sdists if they can but they so often can't (usually because no one can agree on what sdists are for and therefore whether they should include tests/docs) that I don't think I'm really doing anyone any harm in forcing them to use git archives. (I also build Linux packages and have yet to discover why other Linux packagers prefer sdists, which developers rarely test, to git archives.)

I'm not completely opposed to having an sdist but I want someone to tell me that it affects them personally before I even consider it, especially given that there's no way to undo the damage if we put an sdist on PyPI and then decide we'd rather not (short of yanking whole releases).
